sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -62,6 +63,7 @@
      * @param queueName Name of the queue
      */
     QueueAsyncClient(AzureQueueStorageImpl client, String queueName) {
+        Objects.requireNonNull(queueName);
```

Comment:
This check should be in `QueueServiceClient` as it allows for a null queue name to be passed, the builders should already be checking for this.

Reply:
All the entries which asked the queueName will fall into this constructor.
I think it is better to have null checking here and remove the one in buildAsyncClient.

Reply:
While that is true, having it remain in each individually will offer a closer indication to what went wrong and in the case of the client builder it will early out much sooner.